src/aspire/source/simulation.py

```diff
@@ -42,10 +50,33 @@ def __init__(
         noise_filter=None,
     ):
         """
-        A Cryo-EM simulation
-        Other than the base class attributes, it has:
-
-        :param angles: A n-by-3 array of rotation angles
+        A `Simulation` object that supplies images along with other parameters for image manipulation.
+
+        :param L: Resolution of projection images (integer).
+            Default is 8 or `vols.resolution` if a volume is provided.
```

Comment:
Without looking at the code, I think the error message is incorrect (maybe it should refer to `L`, not `self.L`.

There are two ways to view this and it depends if we think simulation should downsample.

If we dont think that, then following #811, I dont think `L` needs to exist either. It woudl come via the vols.

Reply:
for now if using `L` makes the error message sensical, lets change that and cover the rest in 811.

Reply:
Ok, you are correct. The error message should reference `L` instead of `self.L`. I'll make that change, but am leaning towards `L` going away in #811.

Reply:
Fixed the error message and added a note that `L` and `vols.resolution` must agree.